src/core/internal/arrayop.d

```diff
@@ -0,0 +1,428 @@
+        else
+            __builtin_ia32_storedqu(cast(char*) p, val);
+    }
+    else version (DigitalMars)
+    {
+        static if (is(T == float))
+            cast(void) __simd_sto(XMM.STOUPS, *cast(vec*) p, val);
+        else static if (is(T == double))
+            cast(void) __simd_sto(XMM.STOUPD, *cast(vec*) p, val);
+        else
+            cast(void) __simd_sto(XMM.STODQU, *cast(vec*) p, val);
+    }
+}
+
+const(__vector(T[N])) load(T, size_t N)(in T* p)
+{
+    pragma(inline, true);
+    alias vec = __vector(T[N]);
+
+    version (LDC)
+    {
+        return loadUnaligned!vec(cast(T*) p);
+    }
+    else version (GNU)
+    {
+        static if (is(T == float))
+            return __builtin_ia32_loadups(p);
+        else static if (is(T == double))
+            return __builtin_ia32_loadupd(p);
+        else
+            return __builtin_ia32_loaddqu(cast(const char*) p);
+    }
+    else version (DigitalMars)
+    {
+        static if (is(T == float))
+            return __simd(XMM.LODUPS, *cast(const vec*) p);
+        else static if (is(T == double))
+            return __simd(XMM.LODUPD, *cast(const vec*) p);
+        else
+            return __simd(XMM.LODDQU, *cast(const vec*) p);
+    }
+}
+
+const(__vector(T[N])) scalarToVec(T, size_t N)(in T a)
+{
+    pragma(inline, true);
+    alias vec = __vector(T[N]);
+
+    vec res = void;
+    version (DigitalMars) // Bugzilla 7509
+        res.array = [a, a, a, a, a, a, a, a, a, a, a, a, a, a, a, a][0 .. N];
+    else
+        res = a;
+    return res;
+}
+
+__vector(T[N]) binop(string op, T, size_t N)(in __vector(T[N]) a, in __vector(T[N]) b)
+{
+    pragma(inline, true);
+    return mixin("a " ~ op ~ " b");
+}
+
+__vector(T[N]) unaop(string op, T, size_t N)(in __vector(T[N]) a) if (op[0] == 'u')
+{
+    pragma(inline, true);
+    return mixin(op[1 .. $] ~ "a");
+}
+
+// mixin gen
+
+// filter out ops without matching SSE/SIMD instructions (could be composed of several instructions though)
+bool vectorizeableOps(E)(string[] ops)
+{
+    // dfmt off
+    return !(
+        ops.contains("/", "/=") && __traits(isIntegral, E) ||
+        ops.contains("*", "*=") && __traits(isIntegral, E) && E.sizeof != 2 ||
+        ops.contains("%", "%=")
+    );
+    // dfmt on
+}
+
+// filter out things like float[] = float[] / size_t[]
+enum compatibleVecTypes(E, T : T[]) = is(Unqual!T == Unqual!E); // array elem types must be same (maybe add cvtpi2ps)
+enum compatibleVecTypes(E, T) = is(T : E); // scalar must be convertible to target elem type
+enum compatibleVecTypes(E, Types...) = compatibleVecTypes!(E, Types[0 .. $ / 2])
+        && compatibleVecTypes!(E, Types[$ / 2 .. $]);
+
+template vectorizeable(E : E[], Args...)
+{
+    static if (is(vec!E))
+        enum vectorizeable = vectorizeableOps!E([Filter!(not!isType, Args)])
+                && compatibleVecTypes!(E, Filter!(isType, Args));
+    else
+                    enum vectorizeable = false;
```

Comment:
It's a bug in https://github.com/dlang-community/dfmt/issues/286, fixed manually for now.